library/L1_Peripheral/lpc40xx/test/capture_test.cpp

```diff
@@ -0,0 +1,153 @@
+#include <cstdint>
+
+#include "L1_Peripheral/cortex/interrupt.hpp"
+#include "L1_Peripheral/interrupt.hpp"
+#include "L0_Platform/lpc40xx/LPC40xx.h"
+#include "L1_Peripheral/lpc40xx/timer.hpp"
+#include "L1_Peripheral/lpc40xx/capture.hpp"
+#include "L4_Testing/testing_frameworks.hpp"
+#include "utility/enum.hpp"
+#include "utility/status.hpp"
+#include "utility/units.hpp"
+
+
+namespace sjsu::lpc40xx
+{
+EMIT_ALL_METHODS(Capture);
+
+void DummyCallback(void) {}
+
+TEST_CASE("Testing lpc40xx Capture", "[lpc40xx-capture]")
+{
+    LPC_TIM_TypeDef test_timer_register;
+
+    constexpr Capture::CaptureChannelNumber kChannelList[] = {
+      Capture::kCaptureChannel0,
+      Capture::kCaptureChannel1,
+    };
+
+    constexpr Capture::CaptureEdgeMode kModeList[] = {
+      Capture::kCaptureEdgeModeNone,
+      Capture::kCaptureEdgeModeRising,
+      Capture::kCaptureEdgeModeFalling,
+      Capture::kCaptureEdgeModeBoth
+    };
+
+    IsrPointer test_timer_isr = nullptr;
+
+    static Mock<sjsu::Pin> mock_pin0;
+    static Mock<sjsu::Pin> mock_pin1;
+
+    Fake(
+       Method(mock_pin0, SetPinFunction),
+       Method(mock_pin1, SetPinFunction));
+
+    sjsu::Pin & capture0_input_pin = mock_pin0.get();
+    sjsu::Pin & capture1_input_pin = mock_pin1.get();
+
+    static const Capture::CaptureChannelPartial_t kTestTimerPartial =
+    {
+        .timer_register = &test_timer_register,
+        .power_id = SystemController::Peripherals::kTimer0,
+        .irq = IRQn::TIMER0_IRQn,
+        .user_callback = &test_timer_isr,
+        .capture_pin0 = capture0_input_pin,
+        .capture_pin1 = capture1_input_pin
+    };
+
+    const Capture::CaptureChannel_t kTestTimer =
+    {
+        .channel = kTestTimerPartial,
+        .isr = sjsu::lpc40xx::Capture::TimerHandler<kTestTimerPartial>
+    };
+
+    memset(&test_timer_register, 0, sizeof(test_timer_register));
```

Comment:
Can you put this right below the creation of the `test_timer_register`. Makes it easier to see how the registers was configured before it was assigned to something. For example, its referenced in the kTestPartial, then cleared, which may give someone the idea, that if the tests are failing its due to the fact that you did not clear your register data, when you actually did, but they missed it because they read the code top to bottom, then found what they believed to be the actual cause of the problem, when in actuality, this part is fine, and something they did broke the code.

Basically, could result in a red herring.